src/routes/+layout.svelte

```diff
@@ -49,155 +62,139 @@
+				getSettings(),
+				getAppData(oneWeekAgo),
+				getSession().then((res) => {
+					// try to make things faster
+					if (res.status !== 'success') {
+						throw new Error('Could not get session from database');
+					} else {
+						$user = res.user || null;
+						initializeUserSessionData(res.viewMode);
+						if (res.accessToken && $user) {
+							login($user.id, res.accessToken, false);
+						}
+					}
+				})
+			]);
+			if (resSettings.status === 'error') {
+				throw new Error('Could not get settings from database');
 			}
 
-			intervalId = setInterval(refreshAppState, $settings.refreshInterval.default);
-		} else if ($user.status === 'disabled') {
-			popup(
-				'User ' +
-					$user.name +
-					' does not have permission ' +
-					'to access this app; please contact the admin for help'
-			);
-			await callLogout();
-		} else {
-			await callLogout();
-			throw new Error('Unknown user status');
-		}
-	}
+			$settings = resSettings.settings;
+			$buoys = resSettings.buoys;
+			$users = resAppData.usersById!;
 
-	async function initApp() {
-		const resSettings = await getSettings();
-		if (resSettings.status === 'error') {
-			throw new Error('Could not get settings from database');
-		}
-		$settings = resSettings.settings;
-		$buoys = resSettings.buoys;
+			const rsvById: { [id: string]: any } = $reservations.reduce((obj, rsv) => {
+				obj[rsv.id] = rsv;
+				return obj;
+			}, {});
+			(resAppData.reservations || []).forEach((rsv) => {
+				rsvById[rsv.id] = augmentRsv(rsv);
+			});
+			$reservations = Object.values(rsvById).filter((rsv) => rsv.status !== 'canceled');
 
-		const resSession = await getSession();
-		if (resSession.status === 'error') {
-			throw new Error('Could not get session from database');
+			if (!intervalId) {
+				intervalId = setInterval(initApp, $settings.refreshInterval.default);
+			}
+		} catch (error) {
+			console.error(error);
+		} finally {
+			isLoading = false;
 		}
-		$user = resSession.user!;
-		await initFromUser(resSession.viewMode);
 	}
 
-	const getUserFromAuth = (e: any) => {
+	const getUserFromAuth = async (e: any) => {
 		const uid = _.get(e, 'detail.userId', '');
 		const accessToken = _.get(e, 'detail.accessToken', '');
 		if (uid) {
-			console.log("uid:", uid);
+			console.log('uid:', uid);
 			$loginState = 'in';
-			login(uid, accessToken);
+			await login(uid, accessToken, true);
+			await initializeUserSessionData();
 		} else {
 			$loginState = 'out';
 		}
-	}
+	};
 
 	onMount(() => {
 		if (!isFacebook) {
 			// @ts-ignore
 			toast.promise(initApp(), {
 				loading: 'loading...',
-				error: 'Loading error',
+				error: 'Loading error'
 			});
 		}
+
+		return () => {
+			if (intervalId) {
+				clearInterval(intervalId);
+			}
+		};
 	});
 </script>
 
 <Nprogress />
 <Sidebar />
 <div id="app" class="flex px-1 mx-auto w-full">
 	<main class="lg:ml-72 w-full mx-auto">
-		{#if $user && $loginState === 'in'}
+		{#if $loginState === 'in'}
 			<slot />
 		{:else}
```

Comment:
Not sure but I think it might be slightly cleaner to include the ':else if $loginState === "out"' check here in case $loginState === 'pending' then it won't display the login button.

Reply:
just added a spinner